### DISCUSSION TITLE
How to write a test wire:loading?

### DISCUSSION THREAD

**Discussion by akulmehta** [livewire/livewire#8129] [1. Help]
I have a div which should be shown only while the component is loading like:

```blade
    <div
        class="modal-backdrop show fade bg-body-tertiary"
        wire:loading
    >
        <div class="position-absolute top-50 start-50 translate-middle">
            Loading...    
        </div>
    </div>
```

I am trying to write a test to make sure that this div is only visible during wire:loading.

Here is my attempt:

```php
    public function test_showLoadingOverlay_listener()
    {
        $this->actingAs($this->users['author']);
        Livewire::test('inputs.experiment-loading-overlay')
            ->assertDontSee('Loading...')
            ->call('showLoadingOverlay')
            ->assertSee('Loading...');
    }
```

Based on this - the `assertDontSee` should pass as the word Loading should not be visible when the component is not loading anything. But when the component method is called, you should see `Loading...` (albeit transiently). 

However, the above test fails at `assertDontSee` before the method call. 

So is there a way to test the wire:loading part of your code? Any suggestion are appreciated.

**Comment by joshhanley** [comment]
@akulmehta `wire:loading` is a javascript feature, so you can't unit test it. The only way to test it is with Dusk tests, but it's not worth it in my opinion.

@christophrumpel has a Livewire testing package where you can assert that there is a `wire:model` in the rendered html. https://github.com/christophrumpel/missing-livewire-assertions

Maybe something similar could be added for `wire:loading` just to assert you have one in your view?

**Reply by akulmehta** [reply]
Thanks. Yeah I wanted to avoid dusk tests. Will check out your suggestions.
